envconfig_test.go

```diff
@@ -70,6 +75,7 @@ type EmbeddedButIgnored struct {
 
 func TestProcess(t *testing.T) {
 	var s Specification
+	s.NestedInterface = &Anonymous{}
```

Comment:
My use case is that I have a single `struct` used by ~20 micro-services:

```
type Configuration struct {
  Logging struct {
    // syslog settings
    // remote logging settings
    // and few more
  }
  Metrics struct {
    // 10 settings about metrics
  }
  GrpcServer struct {
   // TLS settings
   // Port
   // some other grpc specifics
  }
  // and many other settings that all services shared

  Service interface{} // service specifics
}

// and here goes all code that either load file from yaml, json, other source, set development values, etc
```

and in every microservices:

```
type serviceConfiguration struct {
  BusinessLogicServiceSpecific bool
  Other string
}

func main() {
  config := &generic.Configuration{Service: &serviceConfiguration{}}
  config.LoadJSON("/etc/service.json")
}
```